iMX8Pkg/Library/DfciDeviceIdSupportLibStub/Imx8DeviceIdSupportLibStub.c

```diff
@@ -0,0 +1,163 @@
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+**/
+
+#include <PiDxe.h>
+#include <Library/DfciDeviceIdSupportLib.h>
+#include <Library/BaseLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/DebugLib.h>
+
+/**
+Gets the serial number for this device.
+
+@para[out] SerialNumber - UINTN value of serial number
+
+@return EFI_SUCCESS - SerialNumber has been updated to equal the serial number of the device
+@return EFI_ERROR   - Error getting number
+
+**/
+EFI_STATUS
+EFIAPI
+DfciIdSupportV1GetSerialNumber(
+  OUT UINTN*  SerialNumber
+  ) {
+    DEBUG((DEBUG_INFO, "%a - \n", __FUNCTION__));
+   *SerialNumber = 1;
+   return EFI_SUCCESS;
+}
+
+/**
+ * Get the Manufacturer Name
+ *
+ * @param Manufacturer
+ * @param ManufacturerSize
+ *
+ * It is the callers responsibility to free the buffer returned.
+ *
+ * @return EFI_STATUS EFIAPI
+ */
+EFI_STATUS
+EFIAPI
+DfciIdSupportGetManufacturer (
+    CHAR8   **Manufacturer,
+    UINTN    *ManufacturerSize   OPTIONAL
+  ) {
+    DEBUG((DEBUG_INFO, "%a - \n", __FUNCTION__));
+    CHAR8* name = "NXP";
+    UINTN size = AsciiStrnSizeS(name, 50);
+    *Manufacturer = AllocateCopyPool(size, name);
+    DEBUG((DEBUG_INFO, "%a - \n", name));
+
+    if (ManufacturerSize) {
+      *ManufacturerSize = size;
+    }
+    return EFI_SUCCESS;
+}
+
+/**
+ * Get the ProductName
+ *
+ * @param ProductName
+ * @param ProductNameSize
+ *
+ * It is the callers responsibility to free the buffer returned.
+ *
+ * @return EFI_STATUS EFIAPI
+ */
+EFI_STATUS
+EFIAPI
+DfciIdSupportGetProductName (
+    CHAR8   **ProductName,
+    UINTN    *ProductNameSize  OPTIONAL
+  ) {
+    CHAR8* name = "IMX8";
+    UINTN size = AsciiStrnSizeS(name, 50);
+    *ProductName = AllocateCopyPool(size, name);
+    DEBUG((DEBUG_INFO, "%a - %a\n", __FUNCTION__, *ProductName));
+    if (ProductNameSize) {
+      *ProductNameSize = size;
+    }
+    return EFI_SUCCESS;
+}
+
+/**
+ * Get the SerialNumber
+ *
+ * @param SerialNumber
+ * @param SerialNumberSize
+ *
+ * It is the callers responsibility to free the buffer returned.
+ *
+ * @return EFI_STATUS EFIAPI
+ */
+EFI_STATUS
+EFIAPI
+DfciIdSupportGetSerialNumber (
+    CHAR8   **SerialNumber,
+    UINTN    *SerialNumberSize  OPTIONAL
+  ) {
+    CHAR8* serialNumber = "1";
+    UINTN size = AsciiStrnSizeS(serialNumber, 50);
+    *SerialNumber = AllocateCopyPool(size, serialNumber);
+    DEBUG((DEBUG_INFO, "%a - %a\n", __FUNCTION__, *SerialNumber));
+    if (SerialNumberSize) {
+      *SerialNumberSize = size;
+    }
+    return EFI_SUCCESS;
+}
+
+/**
+ * Get the Uuid
+ *
+ * @param Uuid
+ * @param UuidSize
+ *
+ * It is the callers responsibility to free the buffer returned.
```

Comment:
This returns a textized UUID.  Would it be better to return a UUID structure instead to enforce the UUID nature?  Is there a spec for the DFCI library and a schema for it's values?  I'd like to know what we're supposed to be returning.

Reply:
This is because SMBIOS asks for a CHAR8 UUID
DFCI documentation is [currently ongoing](https://microsoft.github.io/mu/dyn/mu_plus/DfciPkg/Dfci_Feature/); there is also some [SEMM documentation](https://docs.microsoft.com/en-us/surface/enroll-and-configure-surface-devices-with-semm)